pkg/api/controller.go

```diff
@@ -3460,6 +3471,96 @@ func (c *Controller) Commit(w http.ResponseWriter, r *http.Request, body apigen.
 	commitResponse(w, r, newCommit)
 }
 
+func (c *Controller) CommitAsync(w http.ResponseWriter, r *http.Request, body apigen.CommitAsyncJSONRequestBody, repository, branch string, params apigen.CommitAsyncParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.CreateCommitAction,
+			Resource: permissions.BranchArn(repository, branch),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "create_commit_async", r, repository, branch, "")
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, "user not found")
+		return
+	}
+	var metadata map[string]string
+	if body.Metadata != nil {
+		metadata = body.Metadata.AdditionalProperties
+	}
+
+	taskID, err := c.asyncOpsHandler.SubmitCommit(ctx, repository, branch, body.Message, user.Committer(), metadata, body.Date, params.SourceMetarange, swag.BoolValue(body.AllowEmpty), graveler.WithForce(swag.BoolValue(body.Force)))
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	writeResponse(w, r, http.StatusAccepted, apigen.TaskCreation{
+		Id: taskID,
+	})
+}
+
+func (c *Controller) CommitStatus(w http.ResponseWriter, r *http.Request, repository, branch string, params apigen.CommitStatusParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.CreateCommitAction,
+			Resource: permissions.BranchArn(repository, branch),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "create_commit_async_status", r, repository, branch, "")
+	taskID := params.Id
+	status, err := c.asyncOpsHandler.GetCommitStatus(ctx, repository, taskID)
+	if status == nil {
+		writeResponse(w, r, http.StatusInternalServerError, apigen.CommitStatus{
+			TaskId:     taskID,
+			Completed:  true,
+			UpdateTime: time.Now(),
+			Error: &apigen.Error{
+				Message: "failed to get commit status",
+			},
+		})
+		return
+	}
+	c.HandleAPIErrorCallback(ctx, nil, nil, err, catalog.SetTaskStatusCodeAndError(status.Task))
+
+	resp := apigen.CommitStatus{
+		TaskId:    status.Task.Id,
+		Completed: status.Task.Done,
+	}
+	if status.Task.UpdatedAt != nil {
+		resp.UpdateTime = status.Task.UpdatedAt.AsTime()
+	}
+
+	if status.Task.ErrorMsg != "" {
+		resp.Error = &apigen.Error{
+			Message: status.Task.ErrorMsg,
+		}
+	}
+
+	if status.Info != nil {
+		resp.Result = &apigen.Commit{
+			Id:           status.Info.Id,
+			Parents:      status.Info.Parents,
+			Committer:    status.Info.Committer,
+			Message:      status.Info.Message,
+			CreationDate: status.Info.CreationDate.AsTime().Unix(),
+			MetaRangeId:  status.Info.MetaRangeId,
+			Metadata: &apigen.Commit_Metadata{
+				AdditionalProperties: status.Info.Metadata,
+			},
+			Generation: apiutil.Ptr(status.Info.Generation),
+			Version:    apiutil.Ptr(int(status.Info.Version)),
+		}
+	}
+
+	statusCode := getStatusCodeFromTaskStatusCode(status.Task.StatusCode)
```

Comment:
The Task Status shouldn't dictate the HTTP response status. 
If a task has failed it doesn't mean the HTTP response has failed.